CS3219-AY2223S1/cs3219-project-ay2223s1-g52
# Implement authentication and profile page

Implemented:

1. Authentication
- [Frontend] Store jwt token as cookies during login, closes #19 
- [Frontend] Protect routes when unauthorized, #41
- [Backend] Middleware to verify token sent from frontend, closes #40

2. Basic profile page
- Change password, closes #8
- Delete user, closes #7

3. Logout, closes #6

4. Linked login to backend, closes #5 

Note:
- Addition of ```SECRET_TOKEN``` in ```.env```
- Addition of origin frontend domain in ```cors``` ```user-service/index.js``` (line 9)—> i’m not too sure whats the implication of this in production
- Cookie storage can be improved using httpOnly attribute to prevent XSS —> possible to implement in future if needed

TODO:
- Blacklist token to ensure user cannot log in with same token again
- Improve #41
